runtime/src/snapshot_utils.rs

```diff
@@ -536,6 +677,112 @@ where
+            maximum_file_size,
+        )?;
+
+    let (incremental_snapshot_file_size, mut incremental_snapshot_data_file_stream) =
+        if let Some(ref incremental_snapshot_root_file_path) =
+            snapshot_root_paths.incremental_snapshot_root_file_path
+        {
+            let (incremental_snapshot_file_size, incremental_snapshot_data_file_stream) =
+                prepare_deserialize_snapshot_data_file(
+                    incremental_snapshot_root_file_path,
+                    maximum_file_size,
+                )?;
+            (
+                Some(incremental_snapshot_file_size),
+                Some(incremental_snapshot_data_file_stream),
+            )
+        } else {
+            (None, None)
+        };
+
+    let mut snapshot_streams = SnapshotStreams {
+        full_snapshot_stream: &mut full_snapshot_data_file_stream,
+        incremental_snapshot_stream: incremental_snapshot_data_file_stream.as_mut(),
+    };
+    let ret = deserializer(&mut snapshot_streams)?;
+
+    check_deserialize_snapshot_data_file(
+        full_snapshot_file_size,
+        &snapshot_root_paths.full_snapshot_root_file_path,
+        &mut full_snapshot_data_file_stream,
+    )?;
+
+    if let Some(ref incremental_snapshot_root_file_path) =
+        snapshot_root_paths.incremental_snapshot_root_file_path
+    {
+        check_deserialize_snapshot_data_file(
+            incremental_snapshot_file_size.unwrap(),
+            incremental_snapshot_root_file_path,
+            incremental_snapshot_data_file_stream.as_mut().unwrap(),
+        )?;
+    }
+
+    Ok(ret)
+}
+
+/// Before running the deserializer function, perform common operations on the snapshot archive
+/// files, such as checking the file size and opening the file into a stream.
+fn prepare_deserialize_snapshot_data_file<P>(
+    snapshot_root_file_path: P,
+    maximum_file_size: u64,
+) -> Result<(u64, BufReader<File>)>
+where
+    P: AsRef<Path>,
+{
+    let snapshot_file_size = fs::metadata(&snapshot_root_file_path)?.len();
+
+    if snapshot_file_size > maximum_file_size {
+        let error_message =
+            format!(
+            "too large snapshot data file to deserialize: {} has {} bytes (max size is {} bytes)",
+            snapshot_root_file_path.as_ref().display(), snapshot_file_size, maximum_file_size
+        );
+        return Err(get_io_error(&error_message));
+    }
+
+    let snapshot_data_file = File::open(&snapshot_root_file_path)?;
+    let snapshot_data_file_stream = BufReader::new(snapshot_data_file);
+
+    Ok((snapshot_file_size, snapshot_data_file_stream))
+}
+
+/// After running the deserializer function, perform common checks to ensure the snapshot archive
+/// files were consumed correctly.
+fn check_deserialize_snapshot_data_file<P>(
+    file_size: u64,
+    file_path: P,
+    file_stream: &mut BufReader<File>,
+) -> Result<()>
+where
+    P: AsRef<Path>,
+{
+    let consumed_size = file_stream.seek(SeekFrom::Current(0))?;
```

Comment:
I think this is equivalent to `file_stream.stream_position()`

Reply:
Neat! Yeah, the actual code wasn't changed from the original implementation, just moved. I'll make this change though, since I agree it is simpler.

Reply:
Done.